app/components/hoc/with-progress/index.js

```diff
@@ -0,0 +1,15 @@
+import {path} from 'wasmuth'
+import mapStateToProps from '/util/mapStateToProps'
+
+/**
+ * Pass the progress of `request` as a prop named `propName`.
+ * The request is expected to exist in the state.
+ * Progress is a float from 0 - 100 if progress is available, null otherwise.
+ */
+export default (request, {propName = 'progress'} = {}) =>
+  mapStateToProps((state, props) => {
+    const {loaded, total} = path(['requests', request, 'progress'], state) || {}
+    return {
+      [propName]: typeof loaded === 'undefined' ? null : loaded / total
```

Comment:
another (succinct) way to check for undefined (and not just falsy):

`loaded != null ? loaded / total : null`

note it's `!=` and NOT `!===` 

This is allowed by standardJS as it's a decently common pattern

Reply:
oh nice! Yeah I was afraid of doing null comparisons